src/libecalc/presentation/yaml/mappers/component_mapper.py

```diff
@@ -128,6 +131,22 @@ def from_yaml_to_dto(
                 except ValidationError as e:
                     raise DtoValidationError(data=data, validation_error=e) from e
 
+            elif component_type == ComponentType.PUMP_V2:  # TO DOMAIN DIRECTLY ...
+                try:
+                    pump_yaml = YamlPump(**data)
+                    return pump_yaml.to_domain_models(references=self.__references, timesteps=timevector)
```

Comment:
Skip timevector here, and call get_model_for_timestep in EnergyCalculator? Define a protocol with the needed methods for v2 models?